File: dotnet/In3/Crypto/SignatureType.cs
namespace In3.Crypto
{
    /// <summary>
    /// Group of constants to be used along with the methods of <see cref="Crypto.Api" />.
    /// </summary>
    public class SignatureType
    {
        private SignatureType(string value) { Value = value; }

        internal string Value { get; set; }

        /// <summary>
        /// For hashes of the RLP prefixed.
        /// </summary>
        public static SignatureType EthSign { get { return new SignatureType("eth_sign"); } }

        /// <summary>
        /// For data that was signed directly.
        /// </summary>
        public static SignatureType Raw { get { return new SignatureType("raw"); } }

        /// <summary>
        /// For data that was hashed and then signed.
        /// </summary>
        public static SignatureType Hash { get { return new SignatureType("hash"); } }
    }
}

File: java/src/in3/utils/SignatureType.java
package in3.utils;

/*
 * Constants for the type of Signature
 */
public enum SignatureType {
  eth_sign(2),
  /**
   * < add Ethereum Signed Message-Proefix, hash and sign the data
   */
  raw(0),
  /**
   * < sign the data directly
   */
  hash(1);

  /**
   * < hash and sign the data
   */

  public final int val;

  private SignatureType(int val) {
    this.val = val;
  }

  private int getValue() {
    return this.val;
  }

  public static SignatureType getEnum(int enumVal) {
    for (SignatureType v : values())
      if (v.getValue() == enumVal) return v;
    return null;
  }
}
